src/PointwiseFunctions/InitialDataUtilities/InitialGuess.hpp

```diff
@@ -0,0 +1,42 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <pup.h>
+
+#include "Parallel/CharmPupable.hpp"
+
+/// Items related to pointwise analytic data for elliptic solves, such as
```

Comment:
1. I think it'd be better move this documentation on the namespace to some other place (e.g. to `src/PointwiseFunctions/InitialDataUtilities/AnalyticSolution.hpp`)
2. Evolution systems have both 'analytic data' and 'analytic solution' which are different from each other, and I'm a bit worried that the `analytic_data` here used for naming namespace can be a little confusing. One quick solution that I could think of is just using `elliptic::initial_data` as same as evolution systems for storing abstract base classes, though I'm not sure it's a clear naming for elliptic codes.
3. Another worry is about putting these source files into `src/PointwiseFunctions/InitialDataUtilities` since evolution and elliptic source files will be mixed in a single directory, which may be fine for running code but not very pleasant for someone who tries to look at this in the future. I haven't thought about how to organize analytic data (or solution) - related base classes of elliptic systems in our refactoring procedure... :thinking: Maybe moving the files currently in `InitialDataUtilities` into a subdirectory `Evolution`?

Reply:
I agree with your concerns, though I don't have a great idea how to structure things better. @wthrowe what do you think?

Reply:
1. I agree that this isn't a particularly natural place for the documentation, but I don't think `AnalyticSolution.hpp` is any better.  Some parts of the code have files entirely for namespace documentation when there's no obvious location, but that's also always felt awkward to me.
2. I don't know the standard terminology for elliptic problems.
3. I think the long-term plan is to move all the initial data and similar out of the `PointwiseFunctions` directory into the system directories.  That should result in a more natural place for these in some elliptic-specific directory.  Until then, this seems fine.